gateway/src/main/java/com/lyft/data/gateway/handler/QueryIdCachingProxyHandler.java

```diff
@@ -141,11 +144,32 @@ public String rewriteTarget(HttpServletRequest request) {
     return targetLocation;
   }
 
-  protected String extractQueryIdIfPresent(String path, String queryParams) {
+  protected String extractQueryIdIfPresent(HttpServletRequest request) {
+    String path = request.getRequestURI();
+    String queryParams = request.getQueryString();
+    try {
+      String queryText = CharStreams.toString(request.getReader());
+      if (!Strings.isNullOrEmpty(queryText)
+          && queryText.toLowerCase().contains("system.runtime.kill_query")) {
```

Comment:
Why do we want extract query id only if the string contains `system.runtime.kill_query`?

Reply:
Please take a look - https://github.com/lyft/presto-gateway/issues/33

Reply:
My concern here is that adding this condition will omit the following case for a given query: 

```
`queryText.toLowerCase().contains("system.runtime.kill_query") = false` and `!Strings.isNullOrEmpty(queryText) = true` 
```

I don't have much context here, but just want to make sure the above case is handled in the rest of the code?

Reply:
lgtm otherwise :)